### PR TITLE
ref(ts): convert pluginstore

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/pluginsStore.tsx [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
+  updating: Map<string, Plugin>;
+};
+
+const PluginStoreConfig: Reflux.StoreDefinition & PluginStoreInterface = {
+  plugins: null,
+  state: {
+    loading: true,
+    plugins: [],
+    error: null,
+    pageLinks: null,
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/pluginsStore.tsx:23]: wouldn't be better to have it outside as a const? so then we could reuse it here and inside the reset function :)

File: src/sentry/static/sentry/app/stores/pluginsStore.tsx [hunk 2]
```diff
@@ -82,13 +106,13 @@ const PluginsStore = Reflux.createStore({
     this.triggerState();
   },
 
-  onUpdateSuccess(id, _updateObj) {
+  onUpdateSuccess(id: string, _updateObj: Partial<Plugin>) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/pluginsStore.tsx:109]: I noticed that we are usually defining the function types in an interface. Example:

```
type PluginStoreInterface = Reflux.StoreDefinition & {
  onUpdateSuccess: (id: string, _updateObj: Partial<Plugin>) => void;
  ...
 }
```

maybe we should do that here too?

Comment (scefali) [reply src/sentry/static/sentry/app/stores/pluginsStore.tsx:109]: @priscilawebdev will the typings get used anywhere? I don't really see the point of adding types for functions if it doesn't impact type checking at all which it doesn't seem to here.

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/stores/pluginsStore.tsx:109]: yes you're right.. I don't know why we're creating a separate interface... Good point 😉

**Review by priscilawebdev — APPROVED** [review]
good Job 👏

**Pull request merged**